components/availability/src/Availability.svelte

```diff
@@ -139,33 +142,99 @@
-        (chosenSlot) => chosenSlot != selectedSlot,
-      );
+  let eventOrganizer: string = "";
+  let eventParticipant: array = []; // email, name, status
+  let consecutiveSlotsEndTime: Date;
+  $: consecutiveSlotsEndTime;
 
-      return "unselected";
-    }
+  $: if (sortedSlots.length > 1) {
+    consecutiveSlotsEndTime = qetConsecutiveSlotsEndTime(sortedSlots);
+  }
+
+  //#region event query
+  let query: Availability.EventQuery;
+  $: query = {
+    component_id: id,
+    access_token: access_token,
+    calendarIDs: [],
+  };
+  //#region event query
+
+  //#region book consecutive slots as one event
+  function sendTimeSlot(timeSlot: Availability.TimeSlot) {
+    dispatchEvent("timeSlotChosen", { timeSlot });
   }
 
-  function sendTimeSlot(selectedSlot: Availability.TimeSlot) {
-    let start_time = new Date(selectedSlot.start_time);
-    let end_time = new Date(selectedSlot.end_time);
-    const timeslot: Availability.TimeSlot = {
-      start_time,
-      end_time,
+  function setTimeSlot(slots: obj[]): Availability.TimeSlot {
+    const consecutiveSlots = hasConsecutiveSlots(slots);
+    return {
+      start_time: slots[0].start_time,
+      end_time: consecutiveSlots ? consecutiveSlotsEndTime : slots[0].end_time,
     };
-    dispatchEvent("timeSlotChosen", {
-      timeslot,
-    });
   }
+
+  function getSlotsDuration(slots: obj[]): number {
+    return slot_size * slots.length;
+  }
+
+  function qetConsecutiveSlotsEndTime(slots: obj[]) {
+    const startTime = new Date(slots[0].start_time);
+    const slotsDuration = getSlotsDuration(slots);
+    return new Date(
+      startTime.setMinutes(startTime.getMinutes() + slotsDuration),
+    );
+  }
+
+  function hasConsecutiveSlots(slots: obj[]): boolean {
+    if (slots.length <= 1) {
+      return false;
+    } else {
+      const lastSlotEndTime = slots[slots.length - 1].end_time.valueOf();
+      const lastConsecutiveSlotEndTime = consecutiveSlotsEndTime.valueOf();
+      return lastSlotEndTime == lastConsecutiveSlotEndTime ? true : false;
+    }
+  }
+  //#endregion book consecutive slots as one event
+
+  //#region booking event logic for single time slot or consecutive time slots
+  function setSlotsSelectedStatus(slots: obj[]) {
+    slots.forEach((slot) => (slot.selectionStatus = "selected"));
+  }
+
+  function resetSlotSelection(slots: obj[]): [] {
+    return (slots = []);
+  }
+
+  function createEventFromTimeSlot(
+    event: Availability.TimeSlot,
+    query: Availability.eventQuery,
+  ) {
+    EventStore.createEvent(event, query);
+  }
+
+  function setSelectedTimeSlots(selectedSlot: obj): obj[] {
+    return (slotSelection =
+      selectedSlot.selectionStatus === "selected"
+        ? [...slotSelection, selectedSlot]
+        : slotSelection.filter((slot) => slot != selectedSlot));
+  }
+
+  function sortAndSetEvent(slots: obj[]) {
+    const slotsCopy = [...slots];
+    const event = setTimeSlot(slotsCopy);
+    createEventFromTimeSlot(event, query);
```

Comment:
Should this really be default on-click behaviour? I think our default on-click behaviour should simply fire an event returning all selected time slots. This allows a developer to implement whatever behaviour they want (e.g. "select the 3 time slots you are available for", which activates the "book now" button).

Reply:
Related to the above: we should consider a way for a developer to "disable all available" time slots programmatically, if this is not already possible.

cc. @philrenaud @ozsivanov @pooja169usp

Reply:
I'm not invested in it functioning one way or another; I was just told this was the default behaviour for "choose" click action! Happy to change it if that's what others think is best :)

Reply:
> we should consider a way for a developer to "disable all available" time slots programmatically, if this is not already possible.

They can pass a `calendars` object like:

```
calendars = [
  {
    email_address: "person@name.com",
    availability: "busy",
    timeslots: [
      {
        start_time: new Date(new Date().setHours(0, 0, 0, 0)),
        end_time: new Date(new Date().setHours(24, 0, 0, 0)),
      },
    ],
  }
]
```

Would that be sufficient?

Reply:
@philrenaud I was wondering if something as simple as the `allow_booking` prop can enable/disable availability selection. That way we can toggle it based on logic defined in an `on:click` event. 

I just know from past experience, I've had to build booking systems where we ask users to "select the 3 best available times" and after the 3rd selection, all other available time slots would get disabled. So if we can provide some way of achieving that, that would be ideal.

Reply:
Regarding on click behaviour: let's have this current action be "choose"; with `allow_booking` set to `false`, the event should simply be emitted without trying to fire an event.

Reply:
For those reading this at a later time:
- consecutive events are joined into one event with the `start_time` of the first slot and the `end_time` of the second slot automatically
- send all time slots to parent